Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -231,6 +393,24 @@ public bool IsAdmin(string username)
         {
             return _context.Participant.Any(p => p.Username == username && p.IsAdmin == true);
         }
+
+        private string GetCustomUnqiueUsername(string username)
+        {
+            var customSuffix = ".custom";
+            if (_context.Participant.Any((p) => p.Username == username + customSuffix && p.IsCustom == true))
+            {
+                var index = 2;
+                string newUsername = "";
+                do
+                {
+                    newUsername = username + index.ToString() + customSuffix;
+                    index++;
+                } while (_context.Participant.Any((p) => p.Username == newUsername && p.IsCustom == true));
+                return newUsername;
+            }
+            else
+                return username + customSuffix;
+        }
```

Comment:
You can greatly simplify this logic:

```suggestion
            var newUsername = username;
            var index = 2;
            while (_context.Participant.Any((p) => p.Username == newUsername && p.IsCustom == true))
            {
                    newUsername = username + index.ToString();
                    index++;
            }
            return newUsername + CUSTOM_PARTICIPANT_USERNAME_SUFFIX;
```

Reply:
@Kevin-Chu58